services/src/main/java/com/linkedin/kmf/services/ProduceService.java

```diff
@@ -43,71 +43,89 @@
 import java.util.concurrent.atomic.AtomicLong;
 
 
-/**
- * This sets up the producers used by Kafka Monitoring and some of the reported metrics.
- */
 public class ProduceService implements Service {
   private static final Logger LOG = LoggerFactory.getLogger(ProduceService.class);
   private static final String METRIC_GROUP_NAME = "produce-service";
 
   private final String _name;
   private final ProduceMetrics _sensors;
-  private final KMBaseProducer _producer;
-  private final ScheduledExecutorService _executor;
+  private KMBaseProducer _producer;
+  private ScheduledExecutorService _executor;
+  private final ScheduledExecutorService _checkAddPartitionsExecutor;
   private final int _produceDelayMs;
   private final boolean _sync;
   /** This can be updated while running when new partitions are added to the monitored topic. */
   private final ConcurrentMap<Integer, AtomicLong> _nextIndexPerPartition;
-  /** This is the last thing that should be updated after adding new partitions. */
+  /** This is the last thing that should be updated after adding new partition sensors. */
   private final AtomicInteger _partitionNum;
   private final int _recordSize;
   private final String _topic;
   private final String _producerId;
   private final AtomicBoolean _running;
-  private final String _zkConnect;
   private final String _brokerList;
-  private final double _rebalanceThreshold;
+  private final Map _producerPropsOverride;
+  private final String _producerClass;
```

Comment:
It seems unnecessary to have this variable _producerClass. We always use it together with _producerPropsOverride, which already contains _producerClass. Can we remove this variable and simplified the signature of method initializeProducer?

And it seems that the `_brokerList` and `_zkConnect` which were added in pull request https://github.com/linkedin/kafka-monitor/pull/22 doesn't need to be private variable as well, right?